### ISSUE TITLE
Time off

### ISSUE THREAD

**Issue by tomislavmamic** [GeneralMagicio/companybrain#2]
Implement a form for submitting times-off (paid and unpaid) in GM Members.

**Comment by tomislavmamic** [comment]
Members can select start and end date.
They can see their and others days off.
They can separately decide how many paid days they want to take. This can be submitted in a salary report.

**Comment by freshelle** [comment]
If possible, can we have a separate report which can be viewed by members which tracks the total time off days available, used, and unused days. You may refer to the "Tracker" table in this sheet - https://docs.google.com/spreadsheets/d/1_qGBrQ8VqbpP1NMFO8yimAsEmYwkzk_TbyXWQnl1ovo/edit?usp=sharing

**Comment by freshelle** [comment]
Can we add a percentage of the time-off? Some just have time-off for half a day or 1/3 of the day. And also a notes column please

**Comment by nebs-dev** [comment]
i found one issue here - i was able to select start date that was higher than end date. This way "Days" column was a negative number.
![Screenshot-2](https://user-images.githubusercontent.com/11771927/187653782-151ce17f-0013-45a5-9076-ca348cb65b38.png)

**Comment by tomislavmamic** [comment]
@freshelle I have made a change to Days off (now renamed to Time off) page in GM Members:
- each member can now see their time given, used and left. It calculates both days and hours.
- instead of percentages, I made it possible for members to select how many hours in the period have they been off and how many hours of those they want to be paid
- added the notes column



Thanks for noticing, @nebs-dev! Unfortunately I can't add such a restriction in Coda yet. It's not fool proof, so we will all have to learn how to use it. That should be good enough because we are a small group and this is an internal tool.

**Comment by tomislavmamic** [comment]
@freshelle I have made a change to Days off (now renamed to Time off) page in GM Members:
- each member can now see their time given, used and left. It calculates both days and hours.
- instead of percentages, I made it possible for members to select how many hours in the period have they been off and how many hours of those they want to be paid
- added the notes column



Thanks for noticing, @nebs-dev! Unfortunately I can't add such a restriction in Coda yet. It's not fool proof, so we will all have to learn how to use it. That should be good enough because we are a small group and this is an internal tool.

**Comment by freshelle** [comment]
@tomislavmamic - With the updated [PTO policy](https://www.notion.so/giveth/Paid-Time-Off-d55f873ccb8d4a80963748f0e773aab8), hourly contributors get 1 hour PTO for every 13 hours worked. So, “days” won’t be applicable to them in the time-off tab in GM members. 
It’s ok if I manually compute / keep track of the PTO for hourly and just update it every month.

I'm not sure if this will complicate thing, or we just make everything in denominations of "hours" instead of days. Let me know what you think!

**Comment by tomislavmamic** [comment]
Currently I calculate everything from days into hours in Time off use table (under HR Stuff in GM Members). I could enable this in 2 ways:
1. Special column where you can manually input accrued paid hours off
2. Just alter the calculation in the Hours given so it doesn't rely on Days given for part time members, and rather check how many hours did the person work in this year and divide that with 13. The same column would calculate full time members time off as it is now, based on manually inputed Days given.

Which one of these sounds better? They are both easy to do.

**Comment by freshelle** [comment]
I prefer option 1 since there may be some months that an hourly contributor is not qualified for PTO for that month (must be at least 75 hours a month)

**Comment by freshelle** [comment]
<img width="708" alt="image" src="https://user-images.githubusercontent.com/86718721/197157146-ea81b2ea-15b3-4d3a-9de6-28fd037e36f3.png">
Can we use `hours` here instead of days for hourly contributors? If that's not possible, maybe we'll just have hours for everyone including full-time?

**Comment by freshelle** [comment]
@tomislavmamic - any updates on the time off computation? It would be nice to have the computation up to date and see the remaining PTO left for everyone.

- for full-time/fixed contributors, we could have it remain at 20 days or the prorated equivalent for this year, then starting January 2022, we'll have it accrued at 1.66667 PTO days a month which will credit every end of the month. Example, Jan PTO credit of 1.66667 will be added on Jan 31, 2023.
- for hourly contributors, I already made some computation in the `2022 Hourly` tab in this [sheet](https://docs.google.com/spreadsheets/d/1_qGBrQ8VqbpP1NMFO8yimAsEmYwkzk_TbyXWQnl1ovo/edit?usp=sharing). We started entering time reports in August, so previous hours can be found in the sheet. It would also be good to show the `eligibility date` on when contributors can start claiming PTO. 

In Giveth, we have it manually computed [here](https://docs.google.com/spreadsheets/d/1xJZMGOGI_UXyiTqyt3LBBLFPNnBFBhQvOvVYLuAZnNI/edit?usp=sharing) for both full-time and hourly. However, the computation on PTO left should be revisited when transferring in Coda considering the validity.
